adoc/chapters/programming_interface.adoc

```diff
@@ -963,207 +963,237 @@ are handled by throwing synchronous SYCL exceptions.
 ----
 
+[[sec:platform-ctors]]
+==== Constructors
 
-[[table.constructors.platform]]
-.Constructors of the SYCL [code]#platform# class
-// Jon: Dims{6.7cm}{8.3cm}
-[width="100%",options="header",separator="@",cols="65%,35%"]
-|====
-@ Constructor @ Description
-a@
-[source]
+.[apititle]#Default constructor#
+[source,role=synopsis,id=api:platform-ctor]
 ----
 platform()
 ----
-   a@ Constructs a SYCL [code]#platform# instance that is a copy of the
-      [code]#platform# which contains the device returned by
-      [code]#default_selector_v#.
 
-a@
-[source]
+_Effects:_ Constructs a [code]#platform# object that is a copy of the platform
+which contains the device returned by [code]#default_selector_v#.
+
+'''
+
+.[apititle]#Selector constructor#
+[source,role=synopsis,id=api:platform-ctor-selector]
 ----
-template <typename DeviceSelector> explicit platform(const DeviceSelector&)
+template <typename DeviceSelector>
+explicit platform(const DeviceSelector& selector)
 ----
-   a@ Constructs a SYCL [code]#platform# instance that is a copy of the
-      [code]#platform# which contains the device returned by the
-      <<device-selector>> parameter.
 
-|====
+_Constraints:_ The [code]#DeviceSelector# must be a type that satisfies the
+requirements of a <<device-selector>> as defined in <<sec:device-selector>>.
 
+_Effects:_ The [code]#selector# is called once for every <<root-device>> as
+described in <<sec:device-selector>>.
+Constructs a [code]#platform# object that is a copy of the platform which
+contains the device that is selected by [code]#selector#.
 
+'''
 
-[[table.members.platform]]
-.Member functions of the SYCL [code]#platform# class
-[width="100%",options="header",separator="@",cols="65%,35%"]
-|====
-@ Member function @ Description
-a@
-[source]
+[[sec:platform-member-funcs]]
+==== Member functions
+
+.[apidef]#platform::get_backend#
+[source,role=synopsis,id=api:platform-get-backend]
 ----
 backend get_backend() const noexcept
 ----
-   a@ Returns a [code]#backend# identifying the <<backend>> associated
-      with this [code]#platform#.
 
-a@
-[source]
+_Returns:_ The <<backend>> that is associated with this platform.
+
+'''
+
+.[apidef]#platform::get_info#
+[source,role=synopsis,id=api:platform-get-info]
 ----
-template <typename Param> typename Param::return_type get_info() const
+template <typename Param>
+typename Param::return_type get_info() const
 ----
-   a@ Queries this SYCL [code]#platform# for information requested by
-      the template parameter [code]#Param#.
-      The type alias [code]#Param::return_type# must be defined in
-      accordance with the info parameters in <<table.platform.info>> to
-      facilitate returning the type associated with the [code]#Param#
-      parameter.
 
-a@
-[source]
+_Constraints:_ The [code]#Param# must be an information descriptor for the
+platform class.
+
+Each information descriptor specifies the return value and may also specify
+preconditions, exceptions that are thrown, etc.
+
```

Comment:
Do we want to put 
```
The type alias [code]#Param::return_type# must be defined in
      accordance with the info parameters in <<table.platform.info>> to
      facilitate returning the type associated with the [code]#Param#
      parameter.
      ```
      
      
      As a `node`?

Reply:
I think we do not need this paragraph anymore.  The updated format specifies the `return_type` type alias for each information descriptor.  See the synopses in the `platform` class "Information descriptors" section.

Reply:
Good point!

Reply:
Do we need to add a hyperlink to that section here to make it easy to find?